### PR TITLE
RET-3610

### PR DESCRIPTION
**Pull request by cindychmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-3610

### Change description ###

CR - ET3 Response - ability to resubmit an ET3

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by hsjhita — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java [hunk 1]
```diff
@@ -294,6 +294,20 @@ public static List<String> createDynamicListSelection(CaseData caseData) {
...
+    private static boolean isAllowSubmit(RespondentSumType respondent) {
+        if (NO.equals(respondent.getResponseReceived())) {
+            return true;
+        }
+        if (respondent.getExtensionDate() != null) {
+            LocalDate extensionDate = LocalDate.parse(respondent.getExtensionDate());
+            return YES.equals(respondent.getExtensionRequested())
+                && YES.equals(respondent.getExtensionGranted())
+                && extensionDate.isAfter(LocalDate.now())
+                && !YES.equals(respondent.getExtensionResubmitted());
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:306]: might be worth null check all of these fields

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java [hunk 2]
```diff
@@ -347,6 +363,11 @@ public static void addEt3DataToRespondent(CaseData caseData) {
     }
 
     private static RespondentSumType addEt3Data(CaseData caseData, RespondentSumType respondent) {
+        if (YES.equals(respondent.getResponseReceived())
```
Comment (Itay-Kirsh) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:366]: Do we not need to null check these here too?

**Review by Itay-Kirsh — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java (hunk 2 above)
Comment (Itay-Kirsh) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:366]: As discussed privately, anything that only needs to be compared with YES.equals doesn't need null checking

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java [hunk 3]
```diff
@@ -294,6 +294,22 @@ public static List<String> createDynamicListSelection(CaseData caseData) {
         return new ArrayList<>();
     }
 
+    private static boolean isAllowSubmit(RespondentSumType respondent) {
+        if (NO.equals(respondent.getResponseReceived())) {
+            return true;
+        }
+        if (respondent.getExtensionRequested() != null
```
[the remark this answers is not in the document]
Comment (Itay-Kirsh) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:301]: Also as mentioned in the next (resolved) comment, don't need to null check the values that are only compared with YES.equals later.

**Review by Itay-Kirsh — APPROVED** [review]

**Pull request merged**